### PR TITLE
Adding support for CNS, partial support for Prometheus, scaleup

### PR DESCRIPTION
**Pull request by jmencak**
Adding support for CNS, partial support for Prometheus, scaleup, wip.

### CODE REVIEW

File: roles/openshift_on_openstack_scaleup/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  with_items: "{{ debug_information['results'] }}"
+
+# Set the path for the OpenStack scaleup log file.
+- name: Creating the OpenStack scaleup log file variable
+  set_fact:
+    openstack_scaleup_log: "{{ ansible_user_dir }}/openstack_scaleup.log"
+
+# Run the Ansible playbook that creates the OpenStack resources.
+- name: Creating the OpenStack resources
+  shell: "{{ ansible_playbook }} -vvv --user openshift -i inventory {{ openshift_openstack_dir }}/openshift-cluster/provision.yml -e openshift_repos_enable_testing=true 2>&1 > {{ openstack_scaleup_log }}"
```
Comment (mbruzek) [inline roles/openshift_on_openstack_scaleup/tasks/main.yml:47]: What does `openshift_repos_enable_testing=true` do exactly? I saw that in the configuration files and but do not know what it does.

**Comment by mbruzek** [comment]
I have reviewed this code and have no blocking comments or questions.
+1 /lgtm

File: roles/openshift_on_openstack_scaleup/vars/main.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+# A list of regular expressions to search and replace in openshift-ansible all.yml.
+all_yml:
+  - { find: "^openshift_openstack_num_masters.*", replace: "openshift_openstack_num_masters: 3" }
```
Comment (mbruzek) [inline roles/openshift_on_openstack_scaleup/vars/main.yml:4]: A scale up operation only adds nodes. Why do you search and replace the number of masters here?

File: roles/openshift_on_openstack_scaleup/vars/main.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+  - { find: "^openshift_openstack_num_nodes.*", replace: "openshift_openstack_num_nodes: 5" }
+  - { find: "^openshift_openstack_num_infra.*", replace: "openshift_openstack_num_infra: 3" }
```
Comment (mbruzek) [inline roles/openshift_on_openstack_scaleup/vars/main.yml:6]: A scale up operation only adds nodes. Why do you replace the infra here? (These values are __already__ 3 from the openshift_on_openstack role.)

File: roles/openshift_on_openstack_scaleup/tasks/main.yml (hunk 1 above)
Comment (jmencak) [reply roles/openshift_on_openstack_scaleup/tasks/main.yml:47]: I was hoping you knew that. :-) Copied from your playbooks.  I wanted to try running without that.

File: roles/openshift_on_openstack_scaleup/vars/main.yml (hunk 2 above)
Comment (jmencak) [reply roles/openshift_on_openstack_scaleup/vars/main.yml:4]: Good point.  I wanted to have a generic scaleup and only really tested scaleup of worker nodes.  Never tested master scaleup.  Feel free to comment this out.

File: roles/openshift_on_openstack_scaleup/vars/main.yml (hunk 3 above)
Comment (jmencak) [reply roles/openshift_on_openstack_scaleup/vars/main.yml:6]: Again, feel free to remove.

**Pull request merged**